commit 90790d3129cf6f5fe095cf01e37d2d0a89d8dbec
From: Steve Dickson <steved@redhat.com>
Date: Mon, 19 Jun 2017 11:19:55 -0400
Subject: mount.nfs: Use default minor version when -o v4 is specified
 When v4 is specified on the command line the
 default minor version needs to be used.
 
 Signed-off-by: Steve Dickson <steved@redhat.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2049262
Bug: https://lore.kernel.org/all/20170619152526.26260-1-steved@redhat.com/
Origin: upstream, https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commitdiff;h=90790d3129cf6f5fe095cf01e37d2d0a89d8dbec
Last-Update: 2024-01-18

Index: nfs-utils-1.3.4/utils/mount/stropts.c
===================================================================
--- nfs-utils-1.3.4.orig/utils/mount/stropts.c	2024-01-18 22:27:44.444515943 +1300
+++ nfs-utils-1.3.4/utils/mount/stropts.c	2024-01-18 22:27:44.440515883 +1300
@@ -119,14 +119,22 @@
 	if (mi->version.v_mode == V_DEFAULT &&
 		config_default_vers.v_mode != V_DEFAULT) {
 		mi->version.major = config_default_vers.major;
-		mi->version.minor = config_default_vers.minor;
+		if (config_default_vers.v_mode == V_SPECIFIC)
+			mi->version.minor = config_default_vers.minor;
+		else
+			mi->version.minor = NFS_DEFAULT_MINOR;
 		return;
 	}
 
 	if (mi->version.v_mode == V_GENERAL) {
 		if (config_default_vers.v_mode != V_DEFAULT &&
-		    mi->version.major == config_default_vers.major)
-			mi->version.minor = config_default_vers.minor;
+		    mi->version.major == config_default_vers.major) {
+			if (config_default_vers.v_mode == V_SPECIFIC)
+				mi->version.minor = config_default_vers.minor;
+			else
+				mi->version.minor = NFS_DEFAULT_MINOR;
+		} else
+			mi->version.minor = NFS_DEFAULT_MINOR;
 		return;
 	}
 
@@ -745,13 +753,9 @@
 	}
 
 	if (mi->version.v_mode != V_SPECIFIC) {
-		if (mi->version.v_mode == V_GENERAL)
-			snprintf(version_opt, sizeof(version_opt) - 1,
-				"vers=%lu", mi->version.major);
-		else
-			snprintf(version_opt, sizeof(version_opt) - 1,
-				"vers=%lu.%lu", mi->version.major,
-				mi->version.minor);
+		snprintf(version_opt, sizeof(version_opt) - 1,
+			"vers=%lu.%lu", mi->version.major,
+			mi->version.minor);
 
 		if (po_append(options, version_opt) == PO_FAILED) {
 			errno = EINVAL;
